app/helpers/grouping_helper.rb

```diff
@@ -0,0 +1,16 @@
+module GroupingHelper
+  def area_tag(area, *classes)
+    x, y = area.coordinate&.slice(:x, :y)&.values_at(:x, :y) || [0, 0]
+    classes.push('area', area.occupied? ? 'occupied' : 'free')
+    content_tag(:div,
+                nil,
+                title: area.name,
+                :'data-type' => area.type,
```

Comment:
See here the key passed to the client which will be used to build the JSON object.